### PR TITLE
final pillowtop cleanup

### PR DESCRIPTION
**Pull request by snopoke**
@czue @benrudolph @calellowitz 

I think this is the last bit of pillow cleanup I'm going to do for now.
- migrate mvp pillows to new architecture
- remove BasicPillow
- move some code around

### CODE REVIEW

File: custom/_legacy/mvp_docs/pillows.py [hunk 1]
```diff
@@ -1,32 +1,74 @@
 import logging
+
 from couchdbkit import ResourceNotFound
+
 from casexml.apps.case.models import CommCareCase
+from corehq.apps.change_feed import topics
```
Comment (dimagimon) [inline custom/_legacy/mvp_docs/pillows.py]: F401 'topics' imported but unused

File: custom/_legacy/mvp_docs/pillows.py [hunk 2]
```diff
@@ -1,32 +1,74 @@
...
+from corehq.apps.change_feed.consumer.feed import KafkaChangeFeed
```
Comment (dimagimon) [inline custom/_legacy/mvp_docs/pillows.py]: F401 'KafkaChangeFeed' imported but unused

File: corehq/ex-submodules/pillow_retry/tests.py [hunk 3]
```diff
@@ -205,21 +209,6 @@ def make_error(id, current_attempt, total_attempts):
         docs_to_process = {e.doc_id for e in errors}
         self.assertEqual({'to-process1', 'to-process2'}, docs_to_process)
 
-    def test_include_doc(self):
```
Comment (snopoke) [inline corehq/ex-submodules/pillow_retry/tests.py]: This only applied to subclasses of `BasicPillow`

File: corehq/ex-submodules/pillow_retry/tasks.py [hunk 4]
```diff
@@ -60,11 +60,6 @@ def process_pillow_retry(error_doc_id):
                 return
 
         change = error_doc.change_object
-        if getattr(pillow, 'include_docs', False):
-            try:
-                change.set_document(pillow.get_couch_db().open_doc(change.id))
-            except ResourceNotFound:
-                change.deleted = True
```
Comment (snopoke) [inline corehq/ex-submodules/pillow_retry/tasks.py]: This only applied to subclasses of `BasicPillow`

**Comment by czue** [comment]
build is failing, fyi
```
======================================================================
ERROR: test_incoming (corehq.apps.sms.tests.test_queueing.QueueingTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vendor/lib/python2.7/site-packages/mock.py", line 1216, in patched
    arg = patching.__enter__()
  File "/vendor/lib/python2.7/site-packages/mock.py", line 1273, in __enter__
    self.target = self.getter()
  File "/vendor/lib/python2.7/site-packages/mock.py", line 1393, in <lambda>
    getter = lambda: _importer(target)
  File "/vendor/lib/python2.7/site-packages/mock.py", line 1126, in _importer
    thing = _dot_lookup(thing, comp, import_path)
  File "/vendor/lib/python2.7/site-packages/mock.py", line 1115, in _dot_lookup
    __import__(import_path)
  File "/mnt/commcare-hq-ro/corehq/apps/sms/management/commands/run_sms_queue.py", line 5, in <module>
    from hqscripts.generic_queue import GenericEnqueuingOperation
  File "/mnt/commcare-hq-ro/hqscripts/generic_queue.py", line 15, in <module>
    class GenericEnqueuingOperation(BaseCommand):
  File "/mnt/commcare-hq-ro/hqscripts/generic_queue.py", line 47, in GenericEnqueuingOperation
    @retry_on_connection_failure
NameError: name 'retry_on_connection_failure' is not defined
```

File: corehq/ex-submodules/pillowtop/listener.py [hunk 5]
```diff
@@ -58,158 +58,6 @@ def lock_manager(obj):
         return LockManager(obj, None)
 
 
-class BasicPillow(PillowBase):
```
Comment (czue) [inline corehq/ex-submodules/pillowtop/listener.py]: :skull: :tada:

File: custom/_legacy/mvp_docs/pillows.py [hunk 6]
```diff
@@ -16,13 +16,19 @@
...
     PillowCheckpointEventHandler
 from pillowtop.checkpoints.util import get_machine_id
-from pillowtop.listener import lock_manager
 from pillowtop.pillow.interface import ConstructedPillow
 from pillowtop.processors.interface import PillowProcessor
 
 pillow_logging = logging.getLogger("pillowtop")
 
 
+def lock_manager(obj):
```
Comment (czue) [inline custom/_legacy/mvp_docs/pillows.py]: could prefix with `_` to make it even more clear that it's only used here

File: custom/_legacy/mvp_docs/pillows.py [hunk 7]
```diff
@@ -236,7 +237,7 @@ def _get_mvp_indicator_pillow(pillow_id, processor, topic):
     return ConstructedPillow(
         name=pillow_id,
         checkpoint=checkpoint,
-        change_feed=KafkaChangeFeed(topics=[topic], group_id='{}-group'.format(pillow_id)),
+        change_feed=CouchChangeFeed(XFormInstance.get_db(), include_docs=False),
```
Comment (czue) [inline custom/_legacy/mvp_docs/pillows.py]: were you planning on doing this later or just going to leave it on couch?

**Comment by czue** [comment]
no blockers, a couple questions. also, I didn't thoroughly review the MVP pillows to make sure the functionality didn't change, but assumed you just carefully moved stuff around. if you want a closer review I'm guessing maybe should have biyeun take a pass

File: custom/_legacy/mvp_docs/pillows.py (hunk 7 above)
Comment (snopoke) [inline custom/_legacy/mvp_docs/pillows.py]: I was just going to leave it on couch but could be persuaded to migrate to kafka if it means there's more cleanup we can do.

**Comment by snopoke** [comment]
@biyeun could you take a pass. All the mvp tests pass but I'd appreciate a second look from someone more familiar with the code. Mostly just moving things around though.

File: corehq/ex-submodules/pillow_retry/tests.py [hunk 8]
```diff
@@ -78,7 +81,8 @@ def test_get_or_create(self):
         self.assertEqual(get.current_attempt, 2)
         self.assertTrue(message in error.error_traceback)
 
-        new = PillowError.get_or_create({'id': id}, FakePillow1())
+        another_pillow = make_fake_constructed_pillow('FakePillow1', '')
```
Comment (dimagimon) [inline corehq/ex-submodules/pillow_retry/tests.py]: F821 undefined name 'make_fake_constructed_pillow'

**Comment by snopoke** [comment]
:white_check_mark:

**Pull request merged**